qubesbuilder/plugins/build_rpm/scripts/rpmbuildinfo

```diff
@@ -0,0 +1,111 @@
+#!/bin/sh
+
+
+# Copyright (c) 2016  Bernhard M. Wiedemann <bernhard@zq1.de>
+# Copyright (c) 2016  Dennis Gilmore <dennis@ausil.us>
+# Copyright (c) 2016  Simon "HW42" Geiser <hw42@ipsumj.de>
+# Copyright (c) 2016  Wojciech Porczyk <woju@invisiblethingslab.com>
+#
+# All rights reserved.
+#
+# Permission to use, copy, modify, and distribute this software for any purpose
+# with or without fee is hereby granted, provided that the above copyright
+# notice and this permission notice appear in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT OF THIRD PARTY RIGHTS. IN
+# NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+# OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+# OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# Except as contained in this notice, the name of a copyright holder shall not
+# be used in advertising or otherwise to promote the sale, use or other dealings
+# in this Software without prior written authorization of the copyright holder.
+#
+
+set -e
+
+getos()
+{
+    # shellcheck disable=SC1091
+    test -r /etc/os-release && . /etc/os-release
+    if test -n "$BUILD_ORIGIN"
+    then
+        ID="$BUILD_ORIGIN"
+    fi
+    if test -z "$ID"; then
+        ID=$(cat /etc/system-release)
+    fi
+    printf %s "$ID"
+}
+
+if test $# -lt 1
+then
+    echo usage: "$0" SRPM >&2
+    exit 2
+fi
+
+if [ "${1%.src.rpm}" = "$1" ]; then
+    SRPMDIR=$(rpm --eval '%{_srcrpmdir}')
+    SRPM="$SRPMDIR/$(rpm -qp --queryformat '%{SOURCERPM}' "$1")"
+    RPMS="$*"
+else
+    SRPM="$1"
+    RPMS=$(find "$(rpm --eval '%{_rpmdir}')" -name '*.rpm')
+fi
+
+if ! [ -e "$SRPM" ]; then
+    echo "$SRPM does not exist" >&2
+    exit 1
+fi
+
+# Build-Date: not from rpm because that will be $SOURCE_DATE_EPOCH
+# Binary: /usr/src/packages/RPMS/*/*.rpm or equivalent
+cat <<EOF
+Format: 0.2-rpm
+Build-Architecture: $(uname -m)
+Source: $(rpm -qp --queryformat '%{name}' "$SRPM")
+Version: $(rpm -qp --queryformat '%{version}-%{release}' "$SRPM")
+Architecture: $(rpm -qp --queryformat '%{arch}' "$SRPM")
+Binary: $(printf '%s' "$RPMS" | xargs rpm -qp --qf "%{name} ")
+Build-Origin: $(getos)
+Build-Date: $(date -R)
+Build-Path: $(rpm --eval '%{_builddir}')
+EOF
```

Comment:
These command substitutions should be in individual assignments, but this script should be rewritten in Python.

Reply:
Same here.